pkg/deploy/api/types.go

```diff
@@ -441,6 +441,20 @@ type DeploymentConfigRollbackSpec struct {
 	IncludeStrategy bool
 }
 
+// DeploymentRequest is a request to a deployment config for a new deployment.
+type DeploymentRequest struct {
+	unversioned.TypeMeta
+	// Name of the deployment config for requesting a new deployment.
+	Name string
+	// DryRun if true should not persist the updated deployment config in storage.
+	DryRun bool
```

Comment:
I'm concerned that we are going to implement a "dry-run" concept across all APIs at some point, in which case you'll need to deal with it here (if the generic mechanism says dry run, this flag shouldn't be required to match).  But it's fine for now